### ISSUE TITLE
Testing redirection to another domain i redux saga

### ISSUE THREAD

**Issue by aleksandarcrvc** [redux-saga/redux-saga#1509]
Hi I have stuck with testing redux saga which redirect to another domain. Here is the code
```js
export function* loginAsCompanyUser(companyUser){
  try {
    yield put(actions.me.loading(true))
    yield call(api.companyUsers.loginAsUser, companyUser.id)
    yield call(redirectToBase)
  } catch (error){
    yield put(actions.notifications.fireErrorNotification('Unable to login you as due to an error. Please try again.'))
    yield put(actions.me.loading(false))
  }
}

export const redirectToBase = () => window.location.href = 'https://www.google.com'

```

And here is the test

```js
describe('loginAsCompanyUser', () => {

  let gen, withError ,companyUser

  beforeAll(()=>{
    companyUser = {
      id: 1,
      user: {
        first_name: 'John',
        last_name: 'Doe',
      },
    }
    gen = cloneableGenerator(loginAsCompanyUser)(companyUser)
  })

  it('set loader on page', () => {
    expect(gen.next().value).toEqual(
      put(actions.me.loading(true))
    )
  })

  it('should call api to impersonate user', () => {
    expect(gen.next().value).toEqual(
      call(api.companyUsers.loginAsUser, companyUser.id)
    )
  })

  it('should redirect to base app', () => {
    expect(gen.next().value).toEqual(
      call(
        redirectToBase
      )
    )
  })

  it('should remove loader in case of failure', () => {
    withError = gen.clone()
    const error = new Error()
    withError.throw(error)

    expect(withError.next().value).toEqual(
      put(actions.me.loading(false))
    )
  })
})
```

The problem is third `it` when I test whether redirect is called generator function will raise error and it will go to catch statement even if that `it` about redirection is passing as test. 

Because of that if I tried to test catch case with cloned generator it will skip the first line in catch statement(because it was called in last it on try part) and it will return `put(actions.me.loading(false))` for next value.

My question is how to test this specific scenarios when I am redirecting to different domain or how to properly do redirections in redux saga in order to avoid error to be thrown in test.

In summary all test are passing but I am unable to test first line i catch statement which is 
` yield put(actions.notifications.fireErrorNotification('Unable to login you as due to an error. Please try again.'))`
because third `it` is throwing error

**Comment by Andarist** [comment]
To test ur catch logic you should call `gen.throw` when you are still "inside" try block and then just compare yielded value (first one in the catch block) against ur expected effect - then u can follow iterating with `gen.next`
